### ISSUE TITLE
Modal is not shown when adding excluded categories in product sync settings is not successful

### ISSUE THREAD

**Issue by ianlin** [woocommerce/facebook-for-woocommerce#2284]
## 🔎 Isolate the bug

- [x] I have confirmed this occurs in the most recent version of WordPress, WooCommerce, and Facebook for WooCommerce.
- [x] I have confirmed this occurs when only WooCommerce and Facebook for WooCommerce are active and when using a default WordPress or WooCommerce theme.

## ✍️ Describe the bug

When adding an excluded category contains products that are currently synced to Facebook, a modal contains `Exclude Products` button is not popup.

## 🚶‍♀️ Steps to reproduce

Using WC `6.8.2` and the `develop` branch with HEAD commit 8a2438e60d0f9290580a89848b3e60374981b47c.

1. Go to Facebook -> Sync Products.
2. Add an excluded category contains products that are currently synced to Facebook.
3. Click `Save changes`.
4. In the Network tab you can see the response data has `success: false`, and the error message:
    ```
    The categories and/or tags that you have selected to exclude from sync contain products that are currently synced to Facebook.<br/><br/>To exclude these products from the Facebook sync, click Exclude Products. To review the category / tag exclusion settings, click Cancel.
    ```
5. But the modal isn't shown in the page.

https://user-images.githubusercontent.com/914406/187392425-69257188-7c10-4245-b00b-28f3ca532858.mov

## ✔️ Expected behavior

The modal should be shown up so the user can either choose to click `Exclude Products` or `Cancel`.

## 🗃 Logs

<details>
`
### WordPress Environment ###

WordPress address (URL): https://ian-local-test-wp.ngrok.io
Site address (URL): https://ian-local-test-wp.ngrok.io
WC Version: 6.8.2
REST API Version: ✔ 6.8.2
WC Blocks Version: ✔ 8.0.0
Action Scheduler Version: ✔ 3.4.0
Log Directory Writable: ✔
WP Version: 6.0.1
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_GB
External object cache: –

### Server Environment ###

Server Info: nginx/1.21.5
PHP Version: 7.4.25
PHP Post Max Size: 150 MB
PHP Time Limit: 30
PHP Max Input Vars: 1000
cURL Version: 7.61.1
OpenSSL/1.1.1g

SUHOSIN Installed: –
MySQL Version: 5.5.5-10.3.32-MariaDB-1:10.3.32+maria~focal
Max Upload Size: 150 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 6.8.2
WC Database Prefix: wp_
Total Database Size: 16.63MB
Database Data Size: 11.77MB
Database Index Size: 4.86MB
wp_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.09MB + Index: 0.09MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 0.31MB + Index: 0.33MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 0.20MB + Index: 0.17MB + Engine InnoDB
wp_automatewoo_abandoned_carts: Data: 0.02MB + Index: 0.08MB + Engine InnoDB
wp_automatewoo_customers: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_automatewoo_customer_meta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_automatewoo_events: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_automatewoo_guests: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_automatewoo_guest_meta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_automatewoo_logs: Data: 0.09MB + Index: 0.17MB + Engine InnoDB
wp_automatewoo_log_meta: Data: 0.28MB + Index: 0.30MB + Engine InnoDB
wp_automatewoo_queue: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_automatewoo_queue_meta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_automatewoo_referrals: Data: 0.02MB + Index: 0.08MB + Engine InnoDB
wp_automatewoo_referral_advocate_keys: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_automatewoo_referral_invites: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.38MB + Index: 0.44MB + Engine InnoDB
wp_followup_coupons: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_followup_coupon_logs: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_followup_customers: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_followup_customer_carts: Data: 0.03MB + Index: 0.03MB + Engine InnoDB
wp_followup_customer_notes: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_followup_customer_orders: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_followup_email_excludes: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_followup_email_logs: Data: 0.06MB + Index: 0.05MB + Engine InnoDB
wp_followup_email_orders: Data: 0.17MB + Index: 0.09MB + Engine InnoDB
wp_followup_email_tracking: Data: 0.05MB + Index: 0.08MB + Engine InnoDB
wp_followup_followup_history: Data: 0.47MB + Index: 0.05MB + Engine InnoDB
wp_followup_order_categories: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_followup_order_items: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_followup_subscribers: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_followup_subscribers_to_lists: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_followup_subscriber_lists: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_gla_budget_recommendations: Data: 0.22MB + Index: 0.14MB + Engine InnoDB
wp_gla_merchant_issues: Data: 0.13MB + Index: 0.00MB + Engine InnoDB
wp_gla_shipping_rates: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_gla_shipping_times: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_options: Data: 6.05MB + Index: 0.09MB + Engine InnoDB
wp_postmeta: Data: 1.50MB + Index: 0.55MB + Engine InnoDB
wp_posts: Data: 0.27MB + Index: 0.06MB + Engine InnoDB
wp_snippets: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.06MB + Index: 0.03MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.05MB + Index: 0.00MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_points_rewards_user_points: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_points_rewards_user_points_log: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_product_attributes_lookup: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_download_directories: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_rate_limits: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wpforms_tasks_meta: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpmailsmtp_debug_events: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpmailsmtp_tasks_meta: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpstg_queue: Data: 0.02MB + Index: 0.00MB + Engine InnoDB

### Post Type Counts ###

attachment: 27
aw_workflow: 5
certificate: 2
certificate_template: 1
course: 1
follow_up_email: 2
lesson: 4
page: 16
post: 2
product: 34
product_variation: 10
question: 3
quiz: 4
restsplain: 38
revision: 42
shop_coupon: 11
shop_order: 85
shop_order_refund: 14
shop_subscription: 3
tec_tc_attendee: 3
tec_tc_order: 4
tec_tc_ticket: 1
tribe_events: 3
tribe_organizer: 1
tribe_rsvp_attendees: 11
tribe_rsvp_tickets: 1
tribe_venue: 1
tribe_wooticket: 4
wcpf_item: 4
wcpf_project: 1
wp_global_styles: 1

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (8) ###

Code Snippets: by Code Snippets Pro – 3.2.0
Facebook for WooCommerce: by Facebook – 2.6.21
WCS Variables: by WooCommerce – 0.1
Follow-Up Emails: by WooCommerce – 4.9.27
WooCommerce Stripe Gateway: by WooCommerce – 6.4.3
WooCommerce: by Automattic – 6.8.2
WP Crontrol: by John Blackbourn & contributors – 1.12.0
WP Mail SMTP: by WPForms – 3.2.1

### Inactive Plugins (12) ###

AutomateWoo: by WooCommerce – 5.5.18
Best Sellers for WooCommerce: by FantasticPlugins – 1.4
Google Listings and Ads: by WooCommerce – 2.0.3
Hello Dolly: by Matt Mullenweg – 1.7.2
Jetpack: by Automattic – 10.9.1
Jetpack Boost: by Automattic – 1.4.2
Product Filters for WooCommerce: by WooCommerce – 1.3.4
Restsplain: by Robert O'Rourke – 1.0.1
WooCommerce Google Analytics Integration: by WooCommerce – 1.5.13
WooCommerce Payments: by Automattic – 4.4.1 (update to version 4.6.0 is available)
WooCommerce Subscriptions: by WooCommerce – 4.4.0
WP STAGING: by WP-STAGING – 2.9.11

### Settings ###

API Enabled: –
Force SSL: –
Currency: TWD (NT$)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 0
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
subscription (subscription)
variable (variable)
variable subscription (variable-subscription)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –
Enforce Approved Product Download Directories: –

### WC Pages ###

Shop base: #7 - /shop/
Basket: #8 - /cart/
Checkout: #9 - /checkout/
My account: #10 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Storefront
Version: 3.9.1 (update to version 4.1.3 is available)
Author URL: https://woocommerce.com/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build
personally we recommend using a child theme. See: How to create a child theme

WooCommerce Support: ✔

### Templates ###

Overrides: –

### Admin ###

Enabled Features: activity-panels
analytics
coupons
customer-effort-score-tracks
experimental-products-task
experimental-import-products-task
experimental-fashion-sample-products
experimental-product-tour
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
shipping-label-banner
subscriptions
store-alerts
transient-notices
wc-pay-promotion
wc-pay-welcome-page
wc-pay-subscriptions-page

Disabled Features: minified-js
settings

Daily Cron: ✔ Next scheduled: 2022-08-31 07:18:03 +08:00
Options: ✔
Notes: 54
Onboarding: completed

### Action Scheduler ###

Canceled: 1
Oldest: 2022-08-25 12:22:19 +0800
Newest: 2022-08-25 12:22:19 +0800

Complete: 712
Oldest: 2022-08-01 14:02:44 +0800
Newest: 2022-08-30 16:21:30 +0800

Failed: 82
Oldest: 2022-06-14 16:01:11 +0800
Newest: 2022-07-06 18:35:26 +0800

Pending: 8
Oldest: 2022-08-30 18:05:28 +0800
Newest: 2022-09-05 15:47:10 +0800

### Status report information ###

Generated at: 2022-08-30 16:39:40 +08:00
`
</details>

**Comment by rawdreeg** [comment]
Thanks for raising this, @ianlin. I haven't been able to reproduce this using the WC 6.8.2 tag, and the develop branch with HEAD commit https://github.com/woocommerce/facebook-for-woocommerce/commit/8a2438e60d0f9290580a89848b3e60374981b47c.

![Screenshot 2022-08-30 at 12 06 13](https://user-images.githubusercontent.com/4209011/187410258-1fef6b2f-ad94-4efd-98c7-1cef117abfd2.jpg)

The error seems to happen because `WCBackboneModal.View` is not loaded by WC when this runs:

https://github.com/woocommerce/facebook-for-woocommerce/blob/8a2438e60d0f9290580a89848b3e60374981b47c/assets/js/admin/settings-sync.js#L126-L129

Did you install an early build of WC 6.8.2? Unfortunately, I couldn't reproduce the issue by checking out the tag and building the WC plugin locally. Could you check if the assets are built properly on your end?

wp/wp-content/plugins/woocommerce/assets/js/admin/backbone-modal.js

**Comment by ianlin** [comment]
Thanks for clarifying @rawdreeg. Hmm that's weird, I did not build WC locally, instead I installed WC 6.8.2 from my WP Plugins page `wp-admin/plugin-install.php` and search for `WooCommerce`. I can see `wp/wp-content/plugins/woocommerce/assets/js/admin/backbone-modal.js` in the WP plugins directory. Then I tried to install `6.8.0` and `6.7.0` downloaded from https://github.com/woocommerce/woocommerce/releases, and the issue still exist.

Since it looks like it's my local environment's problem, I think it's good to close this issue. I will update the issue if I found the problem or if I have other findings.
